### DISCUSSION TITLE
Object data with no field

### DISCUSSION THREAD

**Discussion by danielbattat** [react-hook-form/react-hook-form#3529] [Q&A]
I have some fields that are not part of my user-facing form. These are other properties of the object.

It would be very helpful to manage these in the react-hook-form state so that onsubmit I get the full data.

I've been able to make it work by including my "hidden" properties inside a controller with a null render:
```
<Controller
  render={() => null}
  name={`installments[${installmentIndex}].dueDate`}
  control={control}
  defaultValue={installment.dueDate}                                       
/>
```

Is this bad idea? I'd prefer not to join the data after the onsubmit as I have a very complex form with multiple field arrays so doing that would be overly complicated.

Thank you

**Comment by bluebill1049** [comment]
I think it's easier to do a custom register at `useEffect`
```
useEffect(() => {
 register('xx')
 setValue('xx', 'whatever');
})
```
